curator/src/convert.py

```diff
@@ -0,0 +1,147 @@
+        action_id = 1
+        actions = { 'actions': {} }
+        for project in legacy_config:
+            if project == '.defaults' or project == '.regex':
+                continue
+            unit = legacy_config[project]['delete']['unit']
+            count = legacy_config[project]['delete']['count']
+            raw_regex = legacy_config[project]['raw_regex']
+            if raw_regex:
+                regex = project
+            else:
+                regex = self.format_regex(project)
+            action = self.generate_action(unit, count, regex, exclude=False)
+            actions['actions'][action_id] = action
+            action_id += 1
+        actions['actions'][action_id] = self.generate_defaults(legacy_config)
+        with open(self.actions_file, 'w') as f:
+            yaml.round_trip_dump(actions, f, default_flow_style=False, explicit_start=True)
+
+    def generate_action(self, unit, count, regex, exclude=False):
+        action = {
+            'action': 'delete_indices',
+            'description': 'auto-generated on {0}'.format(time.strftime('%Y-%m-%d %H:%M')),
+            'options': {
+                'ignore_empty_list': True,
+                'continue_if_exception': False,
+                'timeout_override': os.getenv('CURATOR_TIMEOUT', 300)
+            },
+            'filters': [
+                {
+                    'filtertype': 'pattern',
+                    'kind': 'regex',
+                    'value': SingleQuotedScalarString(regex),
+                    'exclude': exclude
+                },
+                {
+                    'filtertype': 'age',
+                    'source': 'name',
+                    'direction': 'older',
+                    'timestring': '%Y.%m.%d',
+                    'unit': unit,
+                    'unit_count': count
+                }
+            ]
+        }
+        return action
+
+    def generate_defaults(self, legacy_config):
+        defaults_node = legacy_config['.defaults']
+        unit = defaults_node['delete']['unit']
+        count = defaults_node['delete']['count']
+        projects = legacy_config.keys()
+        projects.remove('.defaults')
+        if '.regex' in projects:
+            projects.remove('.regex')
+        regex = '|'.join(
+            map(lambda project: project if legacy_config[project]['raw_regex'] else self.format_regex(project), projects)
+            +
+            map(lambda project: self.format_regex(project), ['.searchguard', '.kibana']))
+        return self.generate_action(unit, count, regex, exclude=True)
+
+    def is_empty_actions_file(self):
+        actions_yaml = None
+        with open(self.actions_file, 'r') as f:
+            actions_string = f.read()
+            actions_yaml = yaml.load(actions_string, Loader=yaml.Loader)
+        if actions_yaml is None or actions_yaml['actions'] is None:
+            return True
+        else:
+            return False
+
+    def format_regex(self, project):
```

Comment:
Do we properly handle other indices like '.operations', '.orphanted?

Reply:
* `.operations` - yes
* `.orphanted` - I have no idea. How are these indices created? Do they contain the date of creation? I think that they should. Then they fit the `project.startswith('.') ` branch.
AFAIK only `.kibana` index 
```
project.kube-service-catalog.a5c2d3ed-121e-11e8-be27-0ebf08f2fae0.2018.02.15
project.logging.32004c38-121e-11e8-be27-0ebf08f2fae0.2018.02.15
.kibana
.operations.2018.02.15
.searchguard.logging-es-data-master-kkrxvdkl
```

See example result:
```
actions:
  1:
    action: delete_indices
    description: auto-generated on 2018-02-14 11:11
    filters:
    - exclude: false
      kind: regex
      filtertype: pattern
      value: '^project\.myapp\..*$'
    - direction: older
      filtertype: age
      source: name
      timestring: '%Y.%m.%d'
      unit_count: 7
      unit: days
    options:
      continue_if_exception: false
      timeout_override: '300'
      ignore_empty_list: true
  2:
    action: delete_indices
    description: auto-generated on 2018-02-14 11:11
    filters:
    - exclude: false
      kind: regex
      filtertype: pattern
      value: '^\.operations\..*$'
    - direction: older
      filtertype: age
      source: name
      timestring: '%Y.%m.%d'
      unit_count: 56
      unit: days
    options:
      continue_if_exception: false
      timeout_override: '300'
      ignore_empty_list: true
  3:
    action: delete_indices
    description: auto-generated on 2018-02-14 11:11
    filters:
    - exclude: true
      kind: regex
      filtertype: pattern
      value: '^project\.myapp\..*$|^\.operations\..*$|^\.searchguard\..*$|^\.kibana$'
    - direction: older
      filtertype: age
      source: name
      timestring: '%Y.%m.%d'
      unit_count: 30
      unit: days
    options:
      continue_if_exception: false
      timeout_override: '300'
      ignore_empty_list: true
```

Reply:
@jcantrill please correct me if I'm wrong